lib/elastic_apm/metrics.rb

```diff
@@ -75,24 +76,34 @@ def running?
         !!@running
       end
 
-      def collect_and_send
-        metricset = Metricset.new(labels: labels, **collect)
-        return if metricset.empty?
+      def get(key)
+        sets.fetch(key)
+      end
 
-        callback.call(metricset)
+      def collect_and_send
+        metricsets = collect
+        metricsets.compact!
+        metricsets.each do |m|
+          callback.call(m)
+        end
       end
 
       def collect
-        MUTEX.synchronize do
```

Comment:
Not sure if this mutex was actually necessary?

Reply:
Did you have the mutex because you were mutating `samples`?

Reply:
Don't remember. But now that we have `reset_on_collect` it might be warranted again? I'll think it over last.